### ISSUE TITLE
Should check if header already exists on response before adding

### ISSUE THREAD

**Issue by jimmcslim** [stevejgordon/CorrelationId#3]
I'm not exactly sure how this is happening, but when an exception are thrown inside my code and ASP.NET MVC Core ends up in the error path I get an exception thrown by CorrelationIdMiddleware when it attempts to add the header which is already present in the response (perhaps the CorrelationIdMiddleware is being run twice?) 

Possibly it should check if already present, and leave alone if so, or update if required (maybe behaviour controlled via a setting).

**Comment by stevejgordon** [comment]
@jimmcslim Thanks for reporting this. I suspect this is an issue caused by the exception handling which re-executes the pipeline. Are you by chance registering app.UseCorrelationId() after your calls to UseExceptionHandler? In that case it will cause the CorrelationId middleware to be run run twice if the pipeline is re-executed.

I'll investigate this properly later. I suspect that in either case performing a check to only add the header to the response if it does not already exist is a good idea so I'll look at doing that.

**Comment by jimmcslim** [comment]
Yes, I definitely was registering the CorrelationIdMiddleware after registering the ExceptionHandler. I've fixed that, but thanks for implementing a check in the middleware anyway.
